### PR TITLE
GEODE-8675: Log at query start when <trace> is enabled

### PR DESCRIPTION
**Pull request by DonalEvans**
Authored-by: Donal Evans <doevans@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [N/A] Have you written or updated unit tests to verify your changes?

- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java [hunk 1]
```diff
@@ -783,6 +783,7 @@ public QueryObserver startTrace() {
       } else {
         queryObserver = qo;
       }
+      this.cache.getLogger().info("Starting query: " + this.queryString);
```
Comment (kohlmu-pivotal) [inline geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java:786]: why would you not use Logger defined https://github.com/apache/geode/blob/d8fc2e12ec02f8b3deeb063396d0060f5e8f3ed7/geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java#L69

**Review by kohlmu-pivotal — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java (hunk 1 above)
Comment (DonalEvans) [reply geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java:786]: I was using the same pattern found elsewhere in the class (specifically in the two `endTrace()` methods) for consistency. The defined logger appears to not be used at all in the class as things stand, but if it's better to do so, I can replace all calls to `this.cache.getLogger()` in the class with `logger`.

Comment (kohlmu-pivotal) [reply geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java:786]: Interesting pattern indeed. Wonder why the Cache would be exposing it's logger like that.
If you want, you could always set the logger in the constructor.

Wonder if there is a real reason why the cache is exposing it's logger this way.

But thank you for amending.

**Review by kohlmu-pivotal — APPROVED** [review]

**Review by karensmolermiller — APPROVED** [review]
The docs portion of this PR looks good to me.  I approve of the docs part. I haven't looked at the code.

**Pull request merged**